extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -180,35 +202,16 @@ private TenantConfigContext getStaticTenantContext(RoutingContext context) {
     }
 
     private String resolveStaticTenantId(RoutingContext context) {
-        String tenantId = context.get(OidcUtils.TENANT_ID_ATTRIBUTE);
-        if (isTenantSetByAnnotation(context, tenantId)) {
-            return tenantId;
-        }
-
-        if (tenantResolver.isResolvable()) {
-            tenantId = tenantResolver.get().resolve(context);
-
-            if (tenantId != null) {
-                return tenantId;
-            }
-        }
-
-        if (pathMatchingTenantResolver != null) {
-            tenantId = pathMatchingTenantResolver.resolve(context);
-
+        for (TenantResolver resolver : staticTenantResolvers) {
```

Comment:
IMHO it is worth keeping an explicit sequence for each resolver. Also, checking the tenant id set via the annotation is also used before the dynamic resolution, I'd like to propose to add an issuer based resolver and handle it explicitly, hope it is reasonable

Reply:
I have answered this in your comment regarding annotation-based tenant selection, please let's discuss it at one place. Spoiler alert: I can do it, but I don't understand it. There are cons.

Reply:
@michalvavrik Let me comment below